python/mlx/nn/layers/dropout.py

```diff
@@ -33,3 +33,45 @@ def __call__(self, x):
         mask = mx.random.bernoulli(self._p_1, x.shape)
 
         return (1 / self._p_1) * mask.astype(x.dtype) * x
+
+
+class Dropout2d(Module):
+    """Randomly zero an entire channel. During training, each channel
+    will be zeroed-out independently with probability p.
+
+    The remaining channels are multiplied with :math:`\frac{1}{1-p}` where
+    :math:`p` is the probability of zeroing an element. This is done so the
+    expected value of a given element will remain the same.
+
+    Args:
+        p (float): The probability to zero a channel
+    """
+
+    def __init__(self, p: float = 0.5):
+        super().__init__()
+
+        if p < 0 or p >= 1:
+            raise ValueError("The dropout probability should be in [0, 1)")
+
+        self._p_1 = 1 - p
+
+    def _extra_repr(self):
+        return f"p={1-self._p_1}"
+
+    def __call__(self, x):
+        if x.ndim not in (3, 4):
+            raise ValueError(
+                f"Received input with {x.ndim} dimensions. Expected 3 or 4 dimensions."
+            )
+
+        if self._p_1 == 1 or not self.training:
+            return x
+
+        mask = mx.random.bernoulli(p=self._p_1, shape=x.shape[:2])
+        if x.ndim == 3:
+            mask = mask[..., None]
+
+        if x.ndim == 4:
+            mask = mask[..., None, None]
+
+        return (1 / self._p_1) * mask.astype(x.dtype) * x
```

Comment:
Why do a cast on the mask here? Shouldn't the `*` automatically upcast to the right type?

Reply:
I think it should, yes, I just saw it was done in the regular Dropout so I also did it.

Reply:
Cool, that's probably an oversight. Do you mind removing them both 😄 ?

Reply:
Of course 👍